collector/zfs_linux.go

```diff
@@ -0,0 +1,88 @@
+package collector
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+const (
+	zfsArcstatsProcpath = "spl/kstat/zfs/arcstats"
+)
+
+func (c *zfsCollector) zfsAvailable() (err error) {
+	file, err := c.openArcstatsFile()
+	if err != nil {
+		file.Close()
```

Comment:
Check out this blog: http://blog.golang.org/defer-panic-and-recover.

You could also just use `os.Stat()` to check for the `/proc` file instead of actually opening it.